Source/PropertyTools.Wpf/DataGrid/Definitions/PropertyDefinition.cs

```diff
@@ -299,7 +301,26 @@ public void UpdateDescriptor(Type itemType)
         /// </summary>
         protected void SetEnumItemsSource()
         {
-            this.ItemsSource = Enum.GetValues(this.PropertyType);
+            var values = new List<object>();
+            var enumType = this.PropertyType;
+            if (enumType != null)
+            {
+                var ult = Nullable.GetUnderlyingType(enumType);
+                if (ult != null)
+                {
+                    // TODO: this does not work properly, do we need to use a converter?

```

Comment:
Have you tried to use my fix, I have commited as the first approach to solve the problem?

From what I can see, no further change is to do to fix the bug.
Please try again and tell me if we have turther problems.
See below.

Regards 
Dietmar

https://github.com/objorke/PropertyTools/pull/101/commits/92f51fa6606c541cab9e39ea1a5b3ebd27c1f6c7

commit 92f51fa6606c541cab9e39ea1a5b3ebd27c1f6c7

         protected void SetEnumItemsSource()
         {
-             this.ItemsSource = Enum.GetValues(this.PropertyType);
+            Type propType = this.PropertyType;
+
+            if (propType.IsGenericType &&
+                propType.GetGenericTypeDefinition() == typeof(Nullable<>))
+            {
+                Type[] typeCol = propType.GetGenericArguments();
+                Type nullableType;
+                if (typeCol.Length > 0)
+                {
+                    nullableType = typeCol[0];
+                    if (nullableType.BaseType == typeof(Enum))
+                    {
+                        this.ItemsSource = Enum.GetValues(nullableType);
+                    }
+                }
+            }
+            else
+            {
+                this.ItemsSource = Enum.GetValues(propType);
+            }
         }